### PR TITLE
Feature/misc necessary methods

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description

FIxed various CRUD bugs and improved CRUD logic. Implemented notification system. Implemented contributor reordering logic.

## How should this be tested?

Every functionality is covered with unit + integration tests. All of this can also be tested from frontend application.

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: src/main/java/rs/teslaris/core/importer/utility/OAIPMHParseUtility.java [hunk 1]
```diff
@@ -7,6 +7,7 @@ public class OAIPMHParseUtility {
 
     public static Integer parseBISISID(String id) {
         var tokens = id.split("\\)");
+        var aaa = Integer.parseInt(tokens[1]);
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/importer/utility/OAIPMHParseUtility.java:10]: please remove this line

File: src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        switch (user.getPreferredLanguage().getLanguageCode()) {
+            case "SR":
+                return new Notification(
+                    "Neko vas je dodao na publikaciju (" +
+                        notificationValues.get("title") +
+                        "), ukoliko ova publikacija nije vaša, možete se ukloniti sa nje.",
+                    notificationValues, NotificationType.ADDED_TO_PUBLICATION, user);
+            default:
+                return new Notification(
+                    "Someone added zou to publication (" +
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java:43]:
```suggestion
                    "Someone added you to a publication (" +
```

File: src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import rs.teslaris.core.model.user.User;
+
+public class NotificationFactory {
+
+    public static Notification contructNewOtherNameDetectedNotification(
+        Map<String, String> notificationValues, User user) {
+        switch (user.getPreferredLanguage().getLanguageCode()) {
+            case "SR":
+                return new Notification(
+                    "Dodati ste na publikaciju sa novim oblikom imena (" +
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java:15]: This is a bad practice to have here i18n messages which are going to UI

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja please check my comments

File: src/main/java/rs/teslaris/core/importer/utility/OAIPMHParseUtility.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/importer/utility/OAIPMHParseUtility.java:10]: done

File: src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java (hunk 3 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java:15]: I will now try to migrate this to use Spring's MessageSource.

Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/util/notificationhandling/NotificationFactory.java]: Done :smile: :rocket:

**Review by chenejac — APPROVED** [review]

**Pull request merged**